### PR TITLE
Adding document visibility feature to the docs

### PR DESCRIPTION
**Pull request by isuruh15**
## Purpose
Adding documentation for the Document Visibility selection feature
Related to https://github.com/wso2/product-apim/issues/7852

## Approach
Front-ported from APIM 2.6.0

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/
 - Ran FindSecurityBugs plugin and verified report? yes/
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? no

## Related PRs
https://github.com/wso2/carbon-apimgt/pull/8668

### CODE REVIEW

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Control API Visibility and Subscription Availability in the Developer Portal
+# Control API Visibility, Subscription Availability and API documentation visibility in the Developer Portal
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:1]: This doc can be moved under the API documentation section?

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 2]
```diff
@@ -123,6 +123,48 @@ When using the REST API directly, the subscription availability options are avai
     </tr>
     </tbody>
     </table>
+    
+    
+## Control API Document Visibility in Developer Portal
+
+By default, any document associated with an API has the same visibility level of the API. That is, if the API is public, 
+its documentation is also visible to all users (registered and anonymous).
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:131]: unnecessary spaces before this line. Please check the same in the entire document.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 3]
```diff
@@ -123,6 +123,48 @@ When using the REST API directly, the subscription availability options are avai
...
+
+!!! note
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:133]: There should be a section "Enabling document visibility in Developer Portal", and the instructions should come under that.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 4]
```diff
@@ -123,6 +123,48 @@ When using the REST API directly, the subscription availability options are avai
...
+add the following configuration.
+
+``` toml
+[apim.publisher]
+enable_api_doc_visibility = "true"
+```
+
+Document visibility levels can be one of the following options:
+
+- **Same as API visibility:** visible to the same user roles who can see the API. For example, if the API's visibility is public, its documentation is visible to all users.
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:147]: these visibility level names do not give the idea of the explained.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 5]
```diff
@@ -123,6 +123,48 @@ When using the REST API directly, the subscription availability options are avai
...
+### Control API document visibility using the API Publisher
+
+1.  Sign in to the API Publisher as an API creator using the following URL: 
+
+     `https://<localhost>:9443/publisher`
+
+2.  [Create a new API]({{base_path}}/learn/design-api/create-api/create-a-rest-api/) or edit an existing API.
+
+3.  In the **Documents** tab, Click **Add New Document**, to see a new drop-down list added to select visibility from. 
+Select the desired visibility level for the document.
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:161]: This should be the next step.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md [hunk 6]
```diff
@@ -123,6 +123,48 @@ When using the REST API directly, the subscription availability options are avai
...
+     
+     [ ![API document visibility selector]({{base_path}}/assets/img/learn/api-doc-visibility-selector.png) ](/{{base_path}}/assets/img/learn/api-doc-visibility-selector.png)
+     [ ![API document visibility selector dropdown]({{base_path}}/assets/img/learn/api-doc-visibility-dropdown.png) ](/{{base_path}}/assets/img/learn/api-doc-visibility-dropdown.png)
+
+4.  Add the Document.
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:166]: What are the visibility levels for the publisher?

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 1 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:1]: Yes we can. I referred APIM 2.6.0 docs when creating this section and there the doc visibility was explained after Subscription availability section.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 6 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:166]: There's no visibility control in publisher. Every user who has access to Publisher portal can see all the documents.

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 5 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md]: fixed in 9234c7c

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 4 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md]: Changed names in 9234c7c .

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 3 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md]: fixed in 9234c7c

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 1 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md:1]: fixed in 9234c7c

File: en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md (hunk 2 above)
Comment (isuruh15) [reply en/docs/learn/design-api/advanced-topics/control-api-visibility-and-subscription-availability-in-developer-portal.md]: fixed in 5ac01b7

**Pull request merged**